### PR TITLE
Added large_bin_attack which has the same affect as unsorted_bin_attack

### PR DESCRIPTION
**Pull request by sajjadium**

### CODE REVIEW

File: glibc_2.25/large_bin_attack.c [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    free(p2);
+    fprintf(stderr, "We free the first and second large chunks now and they will be inserted in the unsorted bin:"
+           " %p --> %p\n\n", (void *)(p2 - 2), (void *)(p1 - 2));
+
+    malloc(0x90);
+    fprintf(stderr, "Now, we allocate a chunk with a size smaller than the freed first large chunk. This will move the"
+            " freed second large chunk into the large bin freelist, use parts of the freed first large chunk for allocation"
+            ", and reinsert the remaining of the freed first large chunk into the unsorted bin:"
+            " %p\n\n", (void *)((char *)p1 + 0x90));
+
```
Comment (m1ghtym0) [inline glibc_2.25/large_bin_attack.c]: Please add a message that you free the third chunk here

File: glibc_2.25/large_bin_attack.c [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    free(p3);
+ 
+    //------------VULNERABILITY-----------
+
+    fprintf(stderr, "Now emulating a vulnerability that can overwrite the freed second large chunk's \"size\""
+            " as well as its \"bk\" and \"bk_nextsize\" pointers\n");
+    fprintf(stderr, "Basically, we decrease the size of the freed second large chunk to force malloc to insert the freed third large chunk"
```
Comment (m1ghtym0) [inline glibc_2.25/large_bin_attack.c]: Please add a reference to the Background section of https://dangokyo.me/2018/04/07/a-revisit-to-large-bin-in-glibc/ as further reference that explains how large-bins are sorted in the glibc

**Comment by m1ghtym0** [comment]
Thanks for adding this attack:-)
Large-bins have really been missing from this repo!

Please add the small things I noted for 2.25 and 2.26.
Thanks!

**Comment by m1ghtym0** [comment]
Oh and even though not ideal, please rebase your changes to the build script to the fixes from #86 
Thanks!

File: glibc_2.25/large_bin_attack.c (hunk 1 above)
Comment (sajjadium) [reply glibc_2.25/large_bin_attack.c]: I'll take care of it. Thanks for catching it.

File: glibc_2.25/large_bin_attack.c (hunk 2 above)
Comment (sajjadium) [reply glibc_2.25/large_bin_attack.c]: I am not sure whether I understood. I don't have a write access to that blog. Shall I add an extra link to the beginning of the large_bin_attack.c?

Comment (sajjadium) [reply glibc_2.25/large_bin_attack.c]: *which explains how large bins are sorted.

**Comment by sajjadium** [comment]
@m1ghtym0 thanks for checking up on the stuff. Glad to know it was helpful.

File: glibc_2.25/large_bin_attack.c (hunk 2 above)
Comment (m1ghtym0) [reply glibc_2.25/large_bin_attack.c]: I just meant smth. like: "For further reference on how large-bins are handled by ptmalloc check the Background section in https://dangokyo.me/2018/04/07/a-revisit-to-large-bin-in-glibc/"

**Comment by sajjadium** [comment]
Not sure how I can rebase the changes to build script. Github "pull request" is weird lol

**Pull request merged**